crates/telio-traversal/src/endpoint_providers/upnp/mod.rs

```diff
@@ -557,11 +632,10 @@ impl<Wg: WireGuard, I: UpnpEpCommands, E: Backoff> Runtime for State<Wg, I, E> {
                 let _ = self.handle_ping_rx(&buff[..len], &addr).await;
             }
             _ = &mut self.upnp_interval => {
-                if self.check_endpoint_candidate().await.is_ok(){
-                    self.upnp_interval =  PinnedSleep::new(self.exponential_backoff.get_backoff(), ());
-                } else {
+                if self.check_endpoint_candidate().await.is_err() {
```

Comment:
I think this is wrong. You treat 100ms timeout on getting a gateway as an error that will trigger a backoff (If I read it correctly). It works now because of the other bug that makes it 100 second timeout instead of 100ms,  but in general this logic seems to be wrong.

Reply:
The triggering of the next backoff on error is intentional, when the search times out (100ms) it gets a `NoIGDGateway` error and the EP check is tried again after the next backoff period.

Even with the 100s timeout, the worst case would be to wait 10s (internal igd::search_gateway() timeout), but as soon as the upnp gw is found (~5ms on tests) it wakes the task and returns it.